training/src/main/java/wolox/training/models/Book.java

```diff
@@ -18,15 +18,15 @@ public Book() {
 
     public Book(Long id, String author, String image, String title, String subtitle, String publisher,
         String year, Integer pages, String isbn) {
-        this.id = id;
-        this.author = author;
-        this.image = image;
-        this.title = title;
-        this.subtitle = subtitle;
-        this.publisher = publisher;
-        this.year = year;
-        this.pages = pages;
-        this.isbn = isbn;
+        setId(id);
```

Comment:
Don't kill me, but the id should never get passed on the constructor, as it'll be generated by JPA. Therefore you can also remove the setter of id.